httpx/_content.py

```diff
@@ -137,12 +139,12 @@ def encode_content(
 def encode_urlencoded_data(
     data: RequestData,
 ) -> Tuple[Dict[str, str], ByteStream]:
-    plain_data = []
+    plain_data: "List[Tuple[str, str]]" = []
     for key, value in data.items():
-        if isinstance(value, (list, tuple)):
-            plain_data.extend([(key, primitive_value_to_str(item)) for item in value])
-        else:
+        if value is None or isinstance(value, PRIMITIVE_DATA_TYPES):
             plain_data.append((key, primitive_value_to_str(value)))
+        else:
+            plain_data.extend([(key, primitive_value_to_str(item)) for item in value])
```

Comment:
Flipping these checks lets us handle _any_ sequence instead of special casing `List` and `Tuple`.